surface_potential_analysis/kernel/kernel.py

```diff
@@ -691,19 +727,18 @@ def get_noise_operators_real_isotropic_stacked(
         rtol=1e-8,
     )
 
-    for axis, n in enumerate(shape):
+    for axis, n in enumerate(shape_operators):
         cloned = data.copy()
         # Build (e^(ikx) +- e^(-ikx)) operators
-        cos_slice = slice_along_axis(slice(1, 1 + n), axis)
-        sin_slice = slice_along_axis(slice(1 + n, None), axis)
-        reverse_slice = slice_along_axis(slice(None, None, -1), axis)
-
-        data[cos_slice] = (
-            cloned[cos_slice] + cloned[sin_slice][reverse_slice]
-        ) / np.sqrt(2)
-        data[sin_slice] = (
-            cloned[cos_slice][reverse_slice] - cloned[sin_slice]
-        ) / np.sqrt(2)
+        end = n // 2
+        cos_slice = slice_along_axis(slice(1, end + 1), axis)
+        conj_cos_slice = slice_along_axis(slice(-1, end - 1, -1), axis)
+
+        sin_slice = slice_along_axis(slice(end + 1, None), axis)
+        conj_sin_slice = slice_along_axis(slice(end - 1, 0, -1), axis)
+
+        data[cos_slice] = (cloned[cos_slice] + cloned[conj_cos_slice]) / np.sqrt(2)
+        data[sin_slice] = (cloned[conj_sin_slice] - cloned[sin_slice]) / np.sqrt(2)
```

Comment:
operators in the form of e^(ikx), so sin is cloned[sin_slice]-cloned[conj]?

Reply:
Yeah it doesn't effect the final noise kernel, but I'll change it to make things more obvious